systemtest/src/main/java/io/strimzi/systemtest/templates/crd/KafkaTemplates.java

```diff
@@ -66,189 +121,247 @@ public static KafkaBuilder kafkaPersistent(String name, int kafkaReplicas, int z
-                        .endValueFrom()
-                    .endJmxPrometheusExporterMetricsConfig()
+                    .withReplicas(kafkaReplicas)
                 .endKafka()
-                .editZookeeper()
-                    .withNewJmxPrometheusExporterMetricsConfig()
-                        .withNewValueFrom()
-                            .withNewConfigMapKeyRef(ZOOKEEPER_METRICS_CONFIG_REF_KEY, metricsConfigMapName, false)
-                        .endValueFrom()
-                    .endJmxPrometheusExporterMetricsConfig()
-                .endZookeeper()
             .endSpec();
+
+        applyDefaultSpecOfKafka(kb, kafkaReplicas);
+        applyDefaultConfigurationOfZookeeperKafka(kb, zookeeperReplicas);
+        applyDefaultLogging(kb, true);
+        applyMemoryRequestsAndLimitsIfNeeded(kb, true);
+
+        return kb;
     }
 
-    private static KafkaBuilder defaultKafka(Kafka kafka, String name, int kafkaReplicas, int zookeeperReplicas) {
+    private static KafkaBuilder defaultKafkaNodePools(Kafka kafka, String clusterName, int kafkaReplicas, int zookeeperReplicas) {
         KafkaBuilder kb = new KafkaBuilder(kafka)
             .withNewMetadata()
-                .withName(name)
-                .withNamespace(kubeClient().getNamespace())
+                .withName(clusterName)
+                .addToAnnotations(Annotations.ANNO_STRIMZI_IO_NODE_POOLS, "enabled")
+            .endMetadata();
+
+        applyDefaultSpecOfKafka(kb, kafkaReplicas);
+        applyDefaultConfigurationOfZookeeperKafka(kb, zookeeperReplicas);
+        applyDefaultLogging(kb, true);
+        applyMemoryRequestsAndLimitsIfNeeded(kb, true);
+        kb = removeFieldsNotRelatedToParticularMode(kb, true);
+
+        return kb;
+    }
+
+    private static KafkaBuilder defaultKafkaKRaft(Kafka kafka, String clusterName, int kafkaReplicas) {
+        KafkaBuilder kb = new KafkaBuilder(kafka)
+            .withNewMetadata()
+                .withName(clusterName)
+                .addToAnnotations(Annotations.ANNO_STRIMZI_IO_NODE_POOLS, "enabled")
+                .addToAnnotations(Annotations.ANNO_STRIMZI_IO_KRAFT, "enabled")
             .endMetadata()
             .editSpec()
                 .editKafka()
-                    .withVersion(Environment.ST_KAFKA_VERSION)
-                    .withReplicas(kafkaReplicas)
-                    .addToConfig("log.message.format.version", TestKafkaVersion.getKafkaVersionsInMap().get(Environment.ST_KAFKA_VERSION).protocolVersion())
+                    .withMetadataVersion(TestKafkaVersion.getKafkaVersionsInMap().get(Environment.ST_KAFKA_VERSION).metadataVersion())
+                .endKafka()
+            .endSpec();
+
+        applyDefaultSpecOfKafka(kb, kafkaReplicas);
+        applyDefaultLogging(kb, false);
+        applyMemoryRequestsAndLimitsIfNeeded(kb, false);
+        kb = removeFieldsNotRelatedToParticularMode(kb, false);
```

Comment:
Why this method returns KafkaBuilder and the above ones no?

Reply:
Because the above methods change the state of KafkaBuilder object but do not set them to `null`. If you set null to for instance ZK without building object it would cause an NPE. Builder does not support nullifying such complex objects I would say.

Reply:
Makes sense, thanks for explanation